subsys/net/lib/sntp/sntp_simple.c

```diff
@@ -65,3 +68,22 @@ int sntp_simple(const char *server, uint32_t timeout, struct sntp_time *time)
 
 	return res;
 }
+
+#if defined(CONFIG_POSIX_CLOCK)
+int sntp_set_system_time(const char *server, uint32_t timeout)
+{
+	struct sntp_time ts;
+	struct timespec tspec;
+	int res = sntp_simple(server, timeout, &ts);
+
+	if (res < 0) {
+		return res;
+	}
+
+	tspec.tv_sec = ts.seconds;
+	tspec.tv_nsec = ((uint64_t)ts.fraction * (1000 * 1000 * 1000)) >> 32;
+	res = clock_settime(CLOCK_REALTIME, &tspec);
```

Comment:
Shouldnt you do something with the result code?

Reply:
Done

Reply:
Although, it only returns a negative number is the ID is not `CLOCK_REALTIME`, so the result will always be positive.